back/src/main/java/org/pknu/weather/service/supports/WeatherRefresher.java

```diff
@@ -0,0 +1,79 @@
+package org.pknu.weather.service.supports;
+
+import static org.pknu.weather.dto.converter.ExtraWeatherConverter.toExtraWeather;
+import static org.pknu.weather.dto.converter.LocationConverter.toLocationDTO;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.pknu.weather.domain.ExtraWeather;
+import org.pknu.weather.domain.Location;
+import org.pknu.weather.domain.Weather;
+import org.pknu.weather.dto.WeatherResponse.ExtraWeatherInfo;
+import org.pknu.weather.feignClient.utils.ExtraWeatherApiUtils;
+import org.pknu.weather.feignClient.utils.WeatherFeignClientUtils;
+import org.pknu.weather.repository.ExtraWeatherRepository;
+import org.pknu.weather.repository.LocationRepository;
+import org.pknu.weather.service.WeatherQueryService;
+import org.pknu.weather.service.WeatherService;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class WeatherRefresher {
```

Comment:
```suggestion 
pbulic class WeatherRefreshService
```
아래 이유들 때문에 해당 클래스는 Service로 분류하는 것은 어떨까 싶습니다. 
- 로직이 Weather & ExtraWeather의 create, update와 관련됨 -> Service 로직과 유사
- 사용자 입력에 의해 동작하지 않음 -> WeatherService에 정의하기엔 애매한 메서드들임

Reply:
원래는 WeatherRefresher 클래스는 유틸리티 폴더에 적합한 유틸리티 클래스라는 생각이 들어서 서비스로 구현하지 않았는데 도메인 객체와 트랜잭션을 관리한다는 점, 날씨 데이터를 최신 상태로 유지하는 목적이 비즈니스 로직과 밀접한 관련이 있다는 생각을 다시하게 되었습니다. 좋은 의견 감사합니다. 반영하도록 하겠습니다.